psh/test-ps.py

```diff
@@ -43,6 +43,9 @@ def harness(p):
                 task = 'psh'
             if task in expected_tasks:
                 expected_tasks.remove(task)
+            # to provide compatibility with older kernel versions, where it's [idle] not [kthread]
+            for task in ('[kthread]', '[idle]'):
+                expected_tasks.remove(task) if task in expected_tasks else expected_tasks
```

Comment:
no assignment here, so the `else` value is not used.

maaybe the simplest solution would be to just allow old synonym
```python
if task == '[idle]'  # allow old name of kernel thread
  task = '[kthread]'
```
(and keep `kthread` only in expected tasks